exp/state/remote_state.go

```diff
@@ -0,0 +1,223 @@
+package state
+
+import (
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/chanzuckerberg/fogg/config"
+	v2 "github.com/chanzuckerberg/fogg/config/v2"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"github.com/hashicorp/terraform/addrs"
+	"github.com/hashicorp/terraform/lang"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+)
+
+// liberal borrowing from https://github.com/hashicorp/terraform-config-inspect/blob/c481b8bfa41ea9dca417c2a8a98fd21bd0399e14/tfconfig/load_hcl.go#L16
+func Run(fs afero.Fs, configFile, path string) error {
+	// figure out which component or account we are talking about
+	conf, err := config.FindAndReadConfig(fs, configFile)
+	if err != nil {
+		return err
+	}
+
+	componentType, componentName, envName, err := conf.PathToComponentType(path)
+	if err != nil {
+		return err
+	}
+	logrus.Debugf("componentType: %s", componentType)
+
+	// collect remote state references
+	references, err := collectRemoteStateReferences(path)
+	if err != nil {
+		return err
+	}
+	logrus.Debugf("in %s found references %#v", path, references)
+
+	// for each reference, figure out if it is an account or component, since those are separate in our configs
+	accounts := []string{}
+	components := []string{}
+
+	// we do accounts for both accounts and env components
+	for _, r := range references {
+		if _, found := conf.Accounts[r]; found {
+			accounts = append(accounts, r)
+		}
+	}
+
+	if componentType == "env" {
+		env := conf.Envs[envName]
+
+		for _, r := range references {
+			if _, found := env.Components[r]; found {
+				components = append(components, r)
+			}
+		}
+	}
+
+	// update fogg.yml with new references
+	logrus.Debugf("found accounts %#v", accounts)
+	logrus.Debugf("found components %#v", components)
+
+	if componentType == "accounts" {
+		if len(accounts) > 0 {
+			c := conf.Accounts[componentName]
+			if c.Common.DependsOn == nil {
+				c.Common.DependsOn = &v2.DependsOn{}
+			}
+
+			c.DependsOn.Accounts = accounts
+			conf.Accounts[componentName] = c
+		}
+	} else if componentType == "env" {
+		if len(accounts) > 0 || len(components) > 0 {
+			c := conf.Envs[envName].Components[componentName]
+
+			if c.Common.DependsOn == nil {
+				c.Common.DependsOn = &v2.DependsOn{}
+			}
+
+			c.DependsOn.Accounts = accounts
+			c.DependsOn.Components = components
+
+			conf.Envs[envName].Components[componentName] = c
+		}
+	}
+
+	return conf.Write(fs, configFile)
+}
+
+func collectRemoteStateReferences(path string) ([]string, error) {
+	fs := tfconfig.NewOsFs()
+
+	references := map[string]bool{}
```

Comment:
https://github.com/chanzuckerberg/go-misc/blob/main/sets/strings.go, either here or in the future